client/my-sites/themes/controller.js

```diff
@@ -125,10 +126,11 @@ export function details( context, next ) {
 		isLoggedIn: !! user
 	};
 
-	context.store.dispatch( setSection( 'themes', {
-		hasSidebar: false,
+	// When we're logged in, we need to remove the sidebar.
+	removeSidebar( context, {
+		section: 'themes',
 		isFullScreen: true
-	} ) );
+	} );
```

Comment:
@nb @ockham 

I made [the PR](https://github.com/Automattic/wp-calypso/pull/3829). Let's continue this there.